tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -395,9 +399,343 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+			scs = append(scs, storageclass)
+		}
+
+		defer func() {
+			ginkgo.By("Delete the SCs created in step 2")
+			for _, sc := range scs {
+				err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		ginkgo.By("Verify the PVCs created in step 3 are bound")
+		pvs, err := fpv.WaitForPVClaimBoundPhase(client, pvcs, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify that the created CNS volumes are compliant and have correct policy id")
+		for i, pv := range pvs {
+			volumeID := pv.Spec.CSI.VolumeHandle
+			storagePolicyMatches, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyNames[i])
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(storagePolicyMatches).To(gomega.BeTrue(), "storage policy verification failed")
+			e2eVSphere.verifyVolumeCompliance(volumeID, true)
+			gomega.Expect(e2eVSphere.verifyDatastoreMatch(volumeID, sharedvmfsURL)).To(
+				gomega.BeTrue(), "volume %v was created on wrong ds", sharedvmfsURL)
+		}
+
+		defer func() {
+			ginkgo.By("Delete the PVCs created in step 3")
+			for i, pvc := range pvcs {
+				err := fpv.DeletePersistentVolumeClaim(client, pvc.Name, namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(pvs[i].Spec.CSI.VolumeHandle)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		ginkgo.By("Create pods with using the PVCs created in step 3 and wait for them to be ready")
+		ginkgo.By("verify we can read and write on the PVCs")
+		pods := createMultiplePods(ctx, client, pvclaims2d, true)
+		defer func() {
+			ginkgo.By("Delete pods")
+			deletePodsAndWaitForVolsToDetach(ctx, client, pods, true)
+		}()
+
+		rand.Seed(time.Now().Unix())
+		testdataFile := fmt.Sprintf("/tmp/testdata_%v_%v", time.Now().Unix(), rand.Intn(1000))
+		ginkgo.By(fmt.Sprintf("Creating a 100mb test data file %v", testdataFile))
+		op, err := exec.Command("dd", "if=/dev/urandom", fmt.Sprintf("of=%v", testdataFile),
+			"bs=1M", "count=100").Output()
+		fmt.Println(op)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			op, err = exec.Command("rm", "-f", testdataFile).Output()
+			fmt.Println(op)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expand pvcs while writing some data on them")
+		var wg sync.WaitGroup
+		currentPvcSize := pvcs[0].Spec.Resources.Requests[v1.ResourceStorage]
+		newSize := currentPvcSize.DeepCopy()
+		newSize.Add(resource.MustParse(diskSize))
+		fsSizes := []int64{}
+
+		for _, pod := range pods {
+			originalSizeInMb, err := getFSSizeMb(f, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			fsSizes = append(fsSizes, originalSizeInMb)
+		}
+		wg.Add(len(pods) * 2)
+		for i, pod := range pods {
+			go writeKnownData2PodInParallel(f, pod, testdataFile, &wg, fsSizes[i]-300)
```

Comment:
Can you please explain why are we taking 300 here?

Reply:
I'm leaving 100 mb for metadata, 100mb for the temp file we are copying and some buffer